semgrep-core/src/engine/Match_search_rules.ml

```diff
@@ -667,8 +667,12 @@ let rec filter_ranges env xs cond =
              in
 
              let env =
-               Eval_generic.bindings_to_env_with_just_strings (fst env.config)
-                 bindings
+               if const_prop then
```

Comment:
I wonder if this option should be independent of the global `constant_propagation` option, maybe we should And them? What should we do if the rule options say `constant_propagation: false` but `metavariable-regex` has `constant-propagation: true`?

Reply:
Actually that's technically already happening, becuse `bindings_to_env_just_strings_const_prop` uses `constant_propagation` from the `config`, but it's definitely confusing to see that.